specs/merge/beacon-chain.md

```diff
@@ -161,16 +170,9 @@ def process_block(state: BeaconState, block: BeaconBlock) -> None:
 
 #### Execution payload processing
 
-##### `get_execution_state`
-
-*Note*: `ExecutionState` class is an abstract class representing Ethereum execution state.
-
-Let `get_execution_state(execution_state_root: Bytes32) -> ExecutionState`  be the function that given the root hash returns a copy of Ethereum execution state. 
-The body of the function is implementation dependent.
-
 ##### `execution_state_transition`
 
-Let `execution_state_transition(execution_state: ExecutionState, execution_payload: ExecutionPayload) -> None` be the transition function of Ethereum execution state. 
+Let `execution_state_transition(execution_state_root: Bytes32, execution_payload: ExecutionPayload, timestamp: uint64) -> None` be the transition function of Ethereum execution state. 
```

Comment:
Any thoughts on including `timestamp` explicitly in the `ExecutionPayload`? Is it valuable to make sure that the "eth1" block can be formed from just the contents of the payload? Or is it okay to require beaconblock context (e.g. convertin the slot to timestamp and passing in separately

Reply:
`timestamp` can be a part of `ExecutionPayload` with a cost of the following validity condition
```python
assert body.execution_payload.timestamp == compute_time_at_slot(state, state.slot)
```

Having this field a part of the payload might be really useful!

Reply:
I like that it remains more self-contained and would default to including it in there unless we have a compelling reason *not* to